packages/typespec-client-generator-core/test/types/array.test.ts

```diff
@@ -164,3 +166,20 @@ it("recursive array type", async () => {
   const prop = testArrayModel.properties[0];
   strictEqual(prop.type, modelProp.type);
 });
+
+it("array with encode", async () => {
```

Comment:
Can we add more test on this?  I think we only support array of string to have this encoding.   Shall we report error/warning if this is used on other types?

Reply:
Link [here](https://github.com/microsoft/typespec/pull/9002#discussion_r2570297016).